lib/yoti/sandbox/sandbox_client.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require_relative 'const'
+
+module Sandbox
+  # Client is responsible for setting up test data in the sandbox instance
+  class Client
+    attr_accessor :app_id
+    attr_accessor :key
+    attr_accessor :base_url
+
+    def initialize(app_id, private_key)
+      @app_id = app_id
+      @base_url = SANDBOX_BASE_URL
+      @key = OpenSSL::PKey::RSA.new(Base64.decode64(private_key)[26..-1])
+    end
+
+    def setup_sharing_profile(profile)
+      endpoint = "/apps/#{app_id}/tokens?\
+nonce=#{SecureRandom.uuid}&timestamp=#{Time.now.to_i}"
+      uri = URI(
+        "#{SANDBOX_BASE_URL}/v1#{endpoint}"
```

Comment:
Not sure we should have /v1 hardcoded in here, it should either be part of the base URL or the endpoint.

Reply:
We don't use the v1 in the sandbox creation request, but we do use the v1 when we're using it. Not sure how to reconcile the different usecases.